sql/mysql_db_structure.sql

```diff
@@ -99,13 +103,13 @@ CREATE TABLE `contest` (
   `endtime_string` varchar(64) NOT NULL COMMENT 'Authoritative absolute or relative string representation of endtime',
   `unfreezetime_string` varchar(64) DEFAULT NULL COMMENT 'Authoritative absolute or relative string representation of unfreezetime',
   `deactivatetime_string` varchar(64) DEFAULT NULL COMMENT 'Authoritative absolute or relative string representation of deactivatetime',
-  `finalizetime` decimal(32,9) DEFAULT NULL COMMENT 'Time when contest was finalized, null if not yet',
+  `finalizetime` decimal(32,9) UNSIGNED DEFAULT NULL COMMENT 'Time when contest was finalized, null if not yet',
   `finalizecomment` text DEFAULT NULL COMMENT 'Comments by the finalizer',
   `b` smallint(3) unsigned NOT NULL default 0 COMMENT 'Number of extra bronze medals',
   `enabled` tinyint(1) unsigned NOT NULL DEFAULT 1 COMMENT 'Whether this contest can be active',
   `starttime_enabled` tinyint(1) unsigned NOT NULL DEFAULT 1 COMMENT 'If disabled, starttime is not used, e.g. to delay contest start',
-  `process_balloons` tinyint(1) UNSIGNED DEFAULT 1 COMMENT 'Will balloons be processed for this contest?',
-  `public` tinyint(1) UNSIGNED DEFAULT 1 COMMENT 'Is this contest visible for the public and non-associated teams?',
+  `process_balloons` tinyint(1) UNSIGNED DEFAULT 1 NOT NULL COMMENT 'Will balloons be processed for this contest?',
```

Comment:
I don't know why it was nullable, did that mean something? Same for public below.

Reply:
I doubt that was on purpose.